libcloud/container/drivers/lxd.py

```diff
@@ -0,0 +1,1269 @@
+    """
+    def __init__(self, name, driver, used_by, config, managed):
+
+        # the name of the storage pool
+        self.name = name
+
+        # the driver (or type of storage pool). e.g. ‘zfs’ or ‘btrfs’, etc.
+        self.driver = driver
+
+        # which containers (by API endpoint /1.0/containers/<name>)
+        # are using this storage-pool.
+        self.used_by = used_by
+
+        # a dictionary with some information about the storage-pool.
+        # e.g. size, source (path), volume.size, etc.
+        self.config = config
+
+        # Boolean that indicates whether LXD manages the pool or not.
+        self.managed = managed
+
+
+class LXDServerInfo(object):
+    """
+    Wraps the response form /1.0
+    """
+
+    @classmethod
+    def build_from_response(cls, metadata):
+
+        server_info = LXDServerInfo()
+        server_info.api_extensions = metadata.get("api_extensions", None)
+        server_info.api_status = metadata.get("api_status", None)
+        server_info.api_version = metadata.get("api_version", None)
+        server_info.auth = metadata.get("auth", None)
+        server_info.config = metadata.get("config", None)
+        server_info.environment = metadata.get("environment", None)
+        server_info.public = metadata.get("public", None)
+        return server_info
+
+    def __init__(self):
+
+        # List of API extensions added after
+        # the API was marked stable
+        self.api_extensions = None
+
+        # API implementation status
+        # (one of, development, stable or deprecated)
+        self.api_status = None
+
+        # The API version as a string
+        self.api_version = None
+
+        # Authentication state,
+        # one of "guest", "untrusted" or "trusted"
+        self.auth = None
+
+        self.config = None
+
+        # Various information about the host (OS, kernel, ...)
+        self.environment = None
+
+        self.public = None
+
+    def __str__(self):
+        return str(self.api_extensions) + str(self.api_status) + \
+            str(self.api_version) + str(self.auth) + str(self.config) + \
+            str(self.environment) + \
+            str(self.public)
+
+
+class LXDResponse(JsonResponse):
+    valid_response_codes = [httplib.OK, httplib.ACCEPTED, httplib.CREATED,
+                            httplib.NO_CONTENT]
+
+    def parse_body(self):
+
+        if len(self.body) == 0 and not self.parse_zero_length_body:
+            return self.body
+
+        try:
+            # error responses are tricky in Docker. Eg response could be
+            # an error, but response status could still be 200
+            content_type = self.headers.get('content-type', 'application/json')
+            if content_type == 'application/json' or content_type == '':
+                if self.headers.get('transfer-encoding') == 'chunked' and \
+                        'fromImage' in self.request.url:
+                    body = [json.loads(chunk) for chunk in
```

Comment:
Eventually it would be nice if we could refactor some of the transfer-encoding chunked logic to a base ``ChunkedEncodingJsonResponse`` or similar Response class.

Reply:
Basically, this is a copy verbatim from the Docker driver which means I should work more on it at some point. PyCharm shows 138 implementations of ```JsonResponse``` we need someone brave :) :)